### ISSUE TITLE
How to use with thumbsup docker?

### ISSUE THREAD

**Issue by StyraxBenzoin** [chaosarium/arxive#2]
Hey there, really nice theme. Do you have an example for how to use this theme with thumbsup docker image? I tried doing `$ git clone https://github.com/chaosarium/PhotoArchive` and mounting the root of the folder as a docker volume and using `--theme-path` flag, e.g.
```
docker run -t                   \
  -v /path/to/photos:/input:ro  \
  -v /path/to/website:/output"   \
  -v /path/to/photoarchive:/mnt/photoarchive"   \
  -u $(id -u):$(id -g)          \
  ghcr.io/thumbsup/thumbsup      \
  thumbsup --input /input --output /output  \
  --theme-path /mnt/photoarchive
``` 

But it didn't seem to work. Seems its not as simple as that!

**Comment by chaosarium** [comment]
Hi! 

I have never tried running the theme in docker before. By your description, is seems like that you are trying to run docker straight from a clone of this git repo. 

I don't believe this repo contains anything that would work with docker. Have you tried using the official thumbsup docker image (https://hub.docker.com/r/thumbsupgallery/thumbsup/) and setting the theme to a clone of this repo? Apprently the thumbsup docker documentation (https://thumbsup.github.io/docs/2-installation/docker/) says you can provide a config file, in which you may be able to set `"theme-path": ".../pathtothisrepo"`

Edit: I don't know if the one on docker hub is up to date. Maybe the docker directory at https://github.com/thumbsup/thumbsup/tree/master/docker is more up to date?

**Comment by StyraxBenzoin** [comment]
Hi, thanks for your reply. I'm using the latest Docker image for [thumbsup](https://github.com/thumbsup/thumbsup/pkgs/container/thumbsup). I read about the [custom themes](https://thumbsup.github.io/docs/4-themes/create/) and mistakenly thought that all I had to do was clone your repo and pass it to docker as a volume and use the `--theme-path file://path/to/theme` flag in `docker run`. But I realise now your theme depends on `moment` which isn't packaged in the tumbsup docker container.

I attempted to manually install the package in the thumbsup docker image and save it as a custom image. The image ran fine but stalled at finding the `moment` package. These were my steps.

```
sudo docker run --name thumbsup -d -i -t ghcr.io/thumbsup/thumbsup /bin/sh
sudo docker exec -it thumbsup sh
npm install --prefix /thumbsup/node_modules/ moment
exit
#Get the running CONTAINER ID
sudo docker ps
# Now commit the change to save as a custom image.
sudo docker commit [CONTAINER ID] thumbsup/custom
sudo docker stop thumbsup

mkdir ~/themes
cd ~/theme
git clone https://github.com/chaosarium/PhotoArchive

sudo docker run -t	\
-v /path/to/photos:/input:ro	\
-v "path/to/website:/output"	\
-v "~/theme/photoarchive:/mnt/photoarchive"	\ # PhotoArchive root folder mounted as docker volume
-u $(id -u):$(id -g)	\
thumbsup/custom	\ # Using the custom image we created that now has the npm moment package
thumbsup --input /input --output /output	\
--theme-path /mnt/photoarchive	\ # Custom theme path inside docker container as defined above

# Truncated thumbsup gallery build here, then;

Unexpected error Cannot find module 'moment'
Require stack:
- /mnt/photoarchive/helpers/date.js
- /thumbsup/node_modules/thumbsup/src/website/theme.js
- /thumbsup/node_modules/thumbsup/src/website/website.js
- /thumbsup/node_modules/thumbsup/src/index.js
- /thumbsup/node_modules/thumbsup/bin/thumbsup.js

┌───────────────────────────────────────────────────────────────────────────┐
│                                                                           │
│  Something went wrong!                                                    │
│                                                                           │
│  An unexpected error occurred and the gallery didn't build successfully.  │
│  This is most likely an edge-case that hasn't been tested before.         │
│                                                                           │
│  Please check the logs at /output/thumbsup.log.                           │
│  To help improve thumbsup and hopefully resolve your problem,             │
│  you can raise an issue at https://github.com/thumbsup/thumbsup/issues.   │
│                                                                           │
└───────────────────────────────────────────────────────────────────────────┘
```

I can go back into the custom container with `sudo docker run --name thumbsup_custom -d -i -t thumbsup/custom /bin/sh` and `sudo docker exec -it thumbsup_custom sh` and navigate to `/thumbsup/node_modules/` and verify with `ls` that `moment` package is definitely still installed there, but there is something else I'm missing for why the container can't find it. 

I'm not very experienced with all this, I'm very much a kind of person that just reads the docs, then try hack it and see. I'm probably at the limit of my ability now, but I don't expect you to go out of your way to find a solution if docker isn't your use case and is not planned!  I'll probably go ask with an issue on the official thumbsup github.

Thanks :)

**Comment by chaosarium** [comment]
Interestingly, the base theme in thumbsup also uses `moment` (https://github.com/thumbsup/thumbsup/blob/master/src/website/theme-base/helpers/date.js). I have the suspicion that it's looking for `moment` in the wrong place... I was able to reproduce your error after moving the theme to a directory outside of the directory I build thumbsup from. One thing that may work is to run `npm install moment` inside where you cloned the theme. I suppose that's `/mnt/photoarchive` in your case. 

This will create a `node_modules` directory inside `photoarchive`, which hopefully contains `moment` that it's not finding.

The tree should look something like this

```
.
├── README.md
├── album.hbs
├── helpers
│   ├── date.js
│   └── monthYear.js
├── node_modules
│   └── moment
├── package-lock.json
├── package.json
├── partials
│   ├── BW-toggle.hbs
│   ├── content.hbs
│   ├── dark-light-toggle.hbs
│   ├── subalbum.hbs
│   └── thumbnail.hbs
├── public
│   ├── jquery.min.js
│   ├── justifiedGallery.min
│   └── lightgallery2
└── theme.less
```
(I never thought about the docker use case and decided to use `moment` for extra date formatting. If date formatting isn't your concern, I think one workaround is to not use anything that depended on `moment`, which was added in commit f6e0a451700e22c1fbe9c00e7d3fe323dd43ff75. If you undo the changes in this commit, things may add up working.)

**Comment by StyraxBenzoin** [comment]
Wow I'm a big dummy. I thought I needed to install moment in the docker container, but all I needed to do was run `npm install moment` in the root of the PhotoArchive theme! I blame my ignorance on how to use `npm`, this is the first time, I'm not a dev. 

So in summary, my original docker run script;
```
docker run -t                   \
  -v /path/to/photos:/input:ro  \
  -v /path/to/website:/output"   \
  -v /path/to/photoarchive:/mnt/photoarchive"   \
  -u $(id -u):$(id -g)          \
  ghcr.io/thumbsup/thumbsup      \
  thumbsup --input /input --output /output  \
  --theme-path /mnt/photoarchive
```
 does build the website successfully now that `moment` is installed in the root folder of PhotoArchive. There was no need to mess around with making a custom docker container at all!

Thank you :)
